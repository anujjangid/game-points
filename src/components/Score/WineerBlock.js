import React from 'react';
import PropTypes from 'prop-types';

import { Column } from '../../styles';
import { List } from './styles';

const WineerBlock = props => {
  const {
    data: { price, game, count }
  } = props;

  return (
    <React.Fragment>
      <List key={price + game}>
        <Column span="4">{game}</Column>
        <Column span="4">{count}</Column>
        <Column span="4">{price}</Column>
      </List>
    </React.Fragment>
  );
};

export default WineerBlock;

WineerBlock.propTypes = {
  data: PropTypes.shape({
    bonus: PropTypes.number,
    count: PropTypes.number.isRequired,
    game: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired
  })
};
