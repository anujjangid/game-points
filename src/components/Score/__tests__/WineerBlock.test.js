import React from 'react';
import { create } from 'react-test-renderer';
import WineerBlock from '../WineerBlock';

const props = {
  data: { price: 10, game: 'A', count: 1 }
};
describe('Test suite for Score Component', () => {
  // create a UI snapshot to indentify UI changes or design regressions

  it('render correctly', () => {
    const component = create(<WineerBlock {...props} />);
    // check if the componenet match with snapshot
    expect(component.toJSON()).toMatchSnapshot();
  });
});
