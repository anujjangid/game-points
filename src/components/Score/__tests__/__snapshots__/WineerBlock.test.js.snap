// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Score Component render correctly 1`] = `
<section
  className="sc-ifAKCX VBDxL"
>
  <div
    className="sc-htpNat ezQThq"
    span="4"
  >
    A
  </div>
  <div
    className="sc-htpNat ezQThq"
    span="4"
  >
    1
  </div>
  <div
    className="sc-htpNat ezQThq"
    span="4"
  >
    10
  </div>
</section>
`;
